### ISSUE TITLE
RTMPS support (Facebook streaming)

### ISSUE THREAD

**Issue by ArtiomShapovalov** [shogo4405/HaishinKit.swift#718]
**Describe the bug**
No stream with facebook rtmps URL and secret key.

**To Reproduce**
1. Go to facebook "live producer"
2. Copy rtmps and secret key
3. See no stream in the "live producer"

**Expected behavior**
Broadcast should appear in the "live producer"
<img width="1385" alt="Screenshot 2020-07-14 at 13 48 29" src="https://user-images.githubusercontent.com/4069820/87417356-b7792880-c5d8-11ea-8e94-0d18de147304.png">

**Comment by floriangbh** [comment]
Hello,

I just tested on my app, everything is working. I'm using `RTMPNWSocket` (see this issue : https://github.com/shogo4405/HaishinKit.swift/issues/708).

**Comment by shogo4405** [comment]
It works for me with `RTMPSocket`.  Please set preference.
```
struct Preference {
    static var defaultInstance = Preference()

    var uri: String? = "rtmps://live-api-s.facebook.com:443/rtmp/"
    var streamName: String? = "Your Stream Key ex. 2586..."
}
```

**Comment by ArtiomShapovalov** [comment]
@floriangbh answer helped

**Comment by Hesham-Salama** [comment]
This doesn't work. 

Pre-steps:

```
if #available(iOS 12.0, *) {
            let params = NWParameters.tls
            rtmpConnection.parameters = params
        }
```
1. Copied url and stream key from Facebook
2. started streaming using my PC's camera
3. playback tab -> tapped above right button. It connects to the server but it doesn't stream.
